mopidy_youtube/backend.py

```diff
@@ -136,57 +41,163 @@ def __init__(self, config, audio):
+    # query is ignored.
+    #
+    # For performance we only do 2 API calls before we reply, one for search
+    # (youtube.API.search) and one to fetch video_count of all playlists
+    # (youtube.Playlist.load_info_mult).
+    #
+    # We also start loading 2 things in the background: info for all videos and
+    # video list for all playlists. Hence, adding search results to the playing
+    # queue will most likely be instantaneous, since all info will be ready by
+    # that time.
+    #
     def search(self, query=None, uris=None, exact=False):
         # TODO Support exact search
+        logger.info("youtube LibraryProvider.search '%s'", query)
 
-        if not query:
-            return
-
-        if 'uri' in query:
-            search_query = ''.join(query['uri'])
-            url = urlparse(search_query)
-            if 'youtube.com' in url.netloc:
-                req = parse_qs(url.query)
-                if 'list' in req:
-                    return SearchResult(
-                        uri='youtube:search',
-                        tracks=resolve_playlist(req.get('list')[0])
-                    )
-                else:
-                    logger.info(
-                        "Resolving Youtube for track '%s'", search_query)
-                    return SearchResult(
-                        uri='youtube:search',
-                        tracks=[resolve_url(search_query)]
-                    )
-        else:
-            search_query = ' '.join(query.values()[0])
-            logger.info("Searching Youtube for query '%s'", search_query)
-            return SearchResult(
-                uri='youtube:search',
-                tracks=search_youtube(search_query)
-            )
+        if not (query and 'any' in query):
+            return None
+
+        search_query = ' '.join(query['any'])
+        logger.info("Searching Youtube for query '%s'", search_query)
+
+        try:
+            entries = youtube.API.search(search_query)
+        except Exception:
+            return None
+
+        # load playlist info (to get video_count) of all playlists together
+        playlists = [e for e in entries if not e.is_video]
+        youtube.Playlist.load_info_mult(playlists)
+
+        tracks = []
+        for entry in entries:
+            if entry.is_video:
+                uri_base = 'youtube:video'
+                album = 'Youtube Video'
+            else:
+                uri_base = 'youtube:playlist'
+                album = 'Youtube Playlist (%s videos)' % entry.video_count
+
+            tracks.append(Track(
+                name=entry.title,
+                comment=entry.id,
+                length=0,
+                artists=[Artist(name=entry.channel)],
+                album=Album(
+                    name=album,
+                    images=entry.thumbnails,
+                ),
+                uri='%s/%s.%s' % (uri_base, safe_url(entry.title), entry.id)
```

Comment:
you mean instead of %'s ? there were lots of %'s in the existing code, should I replace all of them?